server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -648,8 +648,10 @@ public void lock(List<Pool> poolsToLock) {
             "FROM Pool AS Pool " +
             "  INNER JOIN Pool.product AS Product " +
             "  INNER JOIN Product.attributes AS Attribute " +
+            "  LEFT JOIN Pool.entitlements AS Entitlement " +
```

Comment:
Also, @Ceiu mentioned that Hibernate may not send the DISTINCT down to the database (but maybe that is issue only when using Criteria). That would be another good reason not to use JOIN's here at all, but replace them with EXIST subqueries in the WHERE clause.